apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SpannerClientPartial.cs

```diff
@@ -19,12 +19,31 @@ namespace Google.Cloud.Spanner.V1
 {
     public partial class SpannerClient
     {
+        private const string LeaderRoutingHeader = "x-goog-spanner-route-to-leader";
+
         /// <summary>
         /// Returns the effective <see cref="CallSettings"/> used for each API call.
         /// This can be used to create a modified instance of <see cref="CallSettings"/> to pass
         /// for a single API request.
         /// </summary>
         public virtual SpannerSettings Settings { get; protected set; }
+
+        internal void MaybeApplyRouteToLeaderHeader(ref CallSettings settings, TransactionOptions.ModeOneofCase transactionMode)
+        {
+            if (transactionMode == TransactionOptions.ModeOneofCase.ReadWrite ||
+                transactionMode == TransactionOptions.ModeOneofCase.PartitionedDml)
+            {
+                MaybeApplyRouteToLeaderHeader(ref settings);
+            }
+        }
+
+        internal void MaybeApplyRouteToLeaderHeader(ref CallSettings settings)
+        {
+            if (Settings.LeaderRoutingEnabled)
+            {
+                settings = settings.WithHeader(LeaderRoutingHeader, true.ToString());
```

Comment:
I think it would be clearer to put the constant in here, rather than calling `true.ToString()`. (It will also return "True" - I'd expect "true" to be more idiomatic, although I'm not sure the value actually matters.)